infra/gcb/build_and_run_coverage.py

```diff
@@ -0,0 +1,188 @@
+  dockerfile_path = os.path.join(project_dir, 'Dockerfile')
+  name = project_yaml['name']
+  image = project_yaml['image']
+  report_date = datetime.datetime.now().strftime('%Y%m%d')
+
+  build_steps = [
+      {
+          'args': [
+              'clone', 'https://github.com/google/oss-fuzz.git',
+          ],
+          'name': 'gcr.io/cloud-builders/git',
+      },
+      {
+          'name': 'gcr.io/cloud-builders/docker',
+          'args': [
+              'build',
+              '-t',
+              image,
+              '.',
+          ],
+          'dir': 'oss-fuzz/projects/' + name,
+      },
+  ]
+
+  env = CONFIGURATION[:]
+  out = '/workspace/out/' + SANITIZER
+  env.append('OUT=' + out)
+
+  workdir = build_project.workdir_from_dockerfile(dockerfile_path)
+  if not workdir:
+    workdir = '/src'
+
+  # Compilation step.
+  build_steps.append(
+      {
+          'name': image,
+          'env': env,
+          'args': [
+              'bash',
+              '-c',
+               # Remove /out to make sure there are non instrumented binaries.
+               # `cd /src && cd {workdir}` (where {workdir} is parsed from the
+               # Dockerfile). Container Builder overrides our workdir so we need
+               # to add this step to set it back.
+               # Container Builder doesn't pass --rm to docker run yet.
+               'rm -r /out && cd /src && cd {1} && mkdir -p {0} && compile'.format(out, workdir),
+          ],
+      }
+  )
+
+
+  # Split fuzz targets into batches of CORPUS_DOWNLOAD_BATCH_SIZE.
+  for i in xrange(0,  len(fuzz_targets), CORPUS_DOWNLOAD_BATCH_SIZE):
+    download_corpus_args = []
+    for binary_name in fuzz_targets[i : i+CORPUS_DOWNLOAD_BATCH_SIZE]:
+      qualified_name = binary_name
+      if not binary_name.startswith(project_name):
+        qualified_name = '%s_%s' % (project_name, binary_name)
+
+      url = build_project.get_signed_url(
+        CORPUS_BACKUP_URL.format(project_name, qualified_name), method='GET')
+
+      corpus_archive_path = os.path.join('/corpus', binary_name + '.zip')
+      download_corpus_args.append('%s %s' % (corpus_archive_path, url))
+
+    build_steps.append(
+        # Download corpus.
+        {
+            'name': 'gcr.io/oss-fuzz-base/base-runner',
+            'entrypoint': 'download_corpus',
+            'args': download_corpus_args,
+            'volumes': [{'name': 'corpus', 'path': '/corpus'}],
+        }
+    )
+
+  build_steps.extend([
+      # Unpack the corpus and run coverage script.
+      {
+          'name': 'gcr.io/oss-fuzz-base/base-runner',
+          'env': env + ['HTTP_PORT=', 'COVERAGE_EXTRA_ARGS='],
+          'args': [
+              'bash',
+              '-c',
+              'for f in /corpus/*.zip; do unzip -q $f -d ${f%%.*}; done && coverage',
+          ],
+          'volumes': [{'name': 'corpus', 'path': '/corpus'}],
+      },
+      # Upload the report.
+      {
+          'name': 'gcr.io/cloud-builders/gsutil',
```

Comment:
Note to myself: add one more upload step once https://github.com/google/oss-fuzz/pull/1722 lands